Library/ViewModels/PledgeViewModel.swift

```diff
@@ -449,87 +446,115 @@ public class PledgeViewModel: PledgeViewModelType, PledgeViewModelInputs, Pledge
       updateBackingEvent.filter { $0.isTerminating }.mapConst(true)
     )
 
-    // MARK: - Success/Failure Create
+    let submitButtonTapped = Signal.merge(
+      self.submitButtonTappedSignal.mapConst(true),
+      Signal.merge(willUpdateApplePayBacking, willCreateApplePayBacking).mapConst(false)
+    )
 
-    let createPaymentAuthorizationDidFinishSignal = willCreateApplePayBacking
-      .takeWhen(self.paymentAuthorizationDidFinishSignal)
+    // MARK: - Success/Failure
 
-    let createApplePayBackingCompleted = Signal.combineLatest(
-      createApplePayBackingEvent
-        .filter { $0.isTerminating }
-        .ignoreValues(),
-      createPaymentAuthorizationDidFinishSignal
-    )
+    let scaFlowCompletedWithError = self.scaFlowCompletedWithResultSignal
+      .filter { $0.0.status == .failed }
+      .map(second)
+      .skipNil()
 
-    let createApplePayBackingEventSuccess = createApplePayBackingEvent.values()
+    let scaFlowCompletedWithSuccess = self.scaFlowCompletedWithResultSignal
+      .filter { $0.0.status == .succeeded }
+      .map(first)
+      .ignoreValues()
 
-    let createApplePayBackingEventErrors = createApplePayBackingEvent.errors()
-      .map { $0.localizedDescription }
+    let didInitiateApplePayBacking = Signal.merge(
+      willCreateApplePayBacking,
+      willUpdateApplePayBacking
+    )
 
-    let createApplePayBackingError = createApplePayBackingEventErrors
-      .takeWhen(createApplePayBackingCompleted)
+    let paymentAuthorizationDidFinish = didInitiateApplePayBacking
+      .takeWhen(self.paymentAuthorizationDidFinishSignal)
 
-    self.createBackingError = Signal.merge(
-      createApplePayBackingError,
-      createBackingEventError
+    let createOrUpdateApplePayBackingCompleted = Signal.combineLatest(
```

Comment:
Wondering if this should be a `zip` to account for multiple groups of events?:

1. User opens payment sheet and pledges - payment sheet dismisses, but pledge fails.
2. User opens up payment sheet a second time and pledges - because the `createBackingEvent.filter` event has already fired once before, the next `paymentAuthorizationDidFinish` event will cause the `createOrUpdateApplePayBackingCompleted` to fire without waiting for the next request to terminate.

🤔

Reply:
I'll try to write a failing test for it and then prove that scenario 👍

Reply:
Ok @ifbarrera @Scollaco I've taken a new approach which clears the values for the create and update backing events each time a new flow is started. I think this gives us more confidence that we won't have overlapping values between flows being interrupted and new ones starting.

The tests are passing (all several thousand lines of them 😓).

Note right now I don't think the _deprecatedCreateApplePayBacking_ needs this treatment because it's somewhat separate, but I didn't want to spend too much time on that given that we should remove it in the next PR.